src/proposals/proposals.controller.ts

```diff
@@ -578,7 +579,10 @@ export class ProposalsController {
     const proposal = await this.proposalsService.findOne({
       proposalId,
     });
-    if (!proposal) return { canAccess: false };
+
+    if (!proposal) {
+      throw new NotFoundException(`Proposal with ${proposalId} not found`);
+    }
```

Comment:
Probably we should just log a message that the parent proposal is not valid, but return the proposal anyway

Reply:
@nitrosx please check the comment that I left little bit bellow where I explained to Jay why I did this. I think that is the most reasonable way to move forward.